ruby/bps-kafka/lib/bps/kafka.rb

```diff
@@ -0,0 +1,42 @@
+require 'bps'
+require 'kafka'
+
+module BPS
+  module Kafka
+    # ReliablePublisher is a slow, but reliable Kafka Publisher.
+    # There is no batching at all, messages are emitted synchronously.
+    class ReliablePublisher < BPS::Publisher::Abstract
+      DELIVER_MESSAGE_RETRIES = 3 # 1 by kafka-ruby's Kafka::Client default, increase slightly
+
+      class Topic < BPS::Publisher::Topic::Abstract
+        def initialize(client, topic_name)
+          @client = client
+          @topic_name = topic_name
+        end
+
+        def publish(msg_data)
+          @client.deliver_message(msg_data, topic: @topic_name, retries: DELIVER_MESSAGE_RETRIES)
+          nil
+        end
+      end
+
+      def initialize(seed_brokers, client_id: nil)
+        @client = ::Kafka.new(seed_brokers, client_id: client_id)
+      end
+
+      def topic(name)
+        Topic.new(@client, name)
+      end
+
+      def close
+        @client.close
+        nil
+      end
+    end
+  end
+
+  register_publisher('kafka') do |url, opts|
+    addrs = CGI.unescape(url.host).split(',')
+    Kafka::ReliablePublisher.new(addrs, **opts)
```

Comment:
a note for future PR(s):
this should be configurable via URL param, i.e. if we use a reliable vs async publisher, i.e. `mode=sync`

Reply:
I think it's better to do what's done in Go counterpart: `kafka://` vs `kafka+sync://` (multi-scheme registration)

Reply:
I'll keep the `ReliablePublisher` (I don't like this name much, but it's not a SyncProducer, I think as `kafka://` (default impl). At least, for now.